File: app/src/main/java/com/example/smoiapp001/MainViewModel.java
package com.example.smoiapp001;

import android.app.Application;
import android.arch.lifecycle.AndroidViewModel;
import android.arch.lifecycle.LiveData;
import android.util.Log;

import com.example.smoiapp001.models.TransactionEntry;
import com.example.smoiapp001.database.AppDatabase;

import java.util.List;

public class MainViewModel extends AndroidViewModel {

    // Constant for logging
    private static final String TAG = MainViewModel.class.getSimpleName();

    private LiveData<List<TransactionEntry>> transactions;

    public MainViewModel(Application application) {
        super(application);
        AppDatabase database = AppDatabase.getInstance(this.getApplication());
        Log.d(TAG, "Actively retrieving the tasks from the DataBase");
        transactions = database.transactionDao().loadAllTransactions();
    }

    public LiveData<List<TransactionEntry>> getTasks() {
        return transactions;
    }
}

File: app/src/main/java/com/example/smoiapp001/viewmodels/MainViewModel.kt
package com.example.smoiapp001.viewmodels

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import com.example.smoiapp001.MyApplication
import com.example.smoiapp001.database.AppDatabase
import com.example.smoiapp001.database.models.TransactionEntry
import javax.inject.Inject

class MainViewModel() : ViewModel() {

    @Inject
    lateinit var database: AppDatabase
    val transactions: LiveData<List<TransactionEntry>>

    init {
        MyApplication.mAppComponent.inject(this)
        transactions = database.transactionDao().loadAllTransactions()
    }

}
